### ISSUE TITLE
Every time I run psy ls, a new autoclose psy process is started

### ISSUE THREAD

**Issue by TehShrike** [substack/psy#11]
I started psy by running `/opt/iojs/bin/node /opt/iojs/bin/psy daemon --pidfile=/opt/autoexec/psy.pid -l /opt/autoexec/psy.log`

I launched an app by running `/opt/iojs/bin/node /opt/iojs/bin/psy start --pidfile=/opt/autoexec/psy.pid -n communalchecklist.com --cwd=/opt/communal-checklist --env.PORT=8001 -l communalchecklist.log -- node --max_old_space_size=100 server.js `

I run `psy ls` a few times.  Sometimes I see my app in the list, sometimes I don't.

I run `ps aux | grep psy` and see:

```
root     15073  0.4  2.4 667592 25480 ?        Ssl  12:41   0:00 /opt/iojs-v2.3.4-linux-x64/bin/iojs /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/node_modules/auto-daemon/run.js /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/server.js --sockfile /opt/autoexec/sock --autoclose true --parentpid 15064 -- --sockfile /opt/autoexec/sock --pidfile /opt/autoexec/psy.pid --statefile /opt/autoexec/state --autoclose
root     15104  0.6  2.5 667592 26252 ?        Ssl  12:41   0:00 /opt/iojs-v2.3.4-linux-x64/bin/iojs /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/node_modules/auto-daemon/run.js /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/server.js --sockfile /opt/autoexec/sock --autoclose true --parentpid 15095 -- --sockfile /opt/autoexec/sock --pidfile /opt/autoexec/pid --statefile /opt/autoexec/state --autoclose
root     15136  0.6  2.5 667592 25732 ?        Ssl  12:41   0:00 /opt/iojs-v2.3.4-linux-x64/bin/iojs /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/node_modules/auto-daemon/run.js /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/server.js --sockfile /opt/autoexec/sock --autoclose true --parentpid 15122 -- --sockfile /opt/autoexec/sock --pidfile /opt/autoexec/pid --statefile /opt/autoexec/state --autoclose
root     15185  0.6  2.5 667592 25884 ?        Ssl  12:41   0:00 /opt/iojs-v2.3.4-linux-x64/bin/iojs /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/node_modules/auto-daemon/run.js /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/server.js --sockfile /opt/autoexec/sock --autoclose true --parentpid 15176 -- --sockfile /opt/autoexec/sock --pidfile /opt/autoexec/pid --statefile /opt/autoexec/state --autoclose
root     15379  0.8  2.5 667592 25828 ?        Ssl  12:42   0:00 /opt/iojs-v2.3.4-linux-x64/bin/iojs /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/node_modules/auto-daemon/run.js /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/server.js --sockfile /opt/autoexec/sock --autoclose true --parentpid 15370 -- --sockfile /opt/autoexec/sock --pidfile /opt/autoexec/pid --statefile /opt/autoexec/state --autoclose
root     15237  0.0  0.1  13552  2032 pts/1    S+   12:41   0:00 grep psy
```

I run `psy ls` again (and see nothing!)

I run `ps aux | grep psy` again and see:

```
root     15073  0.2  2.4 667592 25480 ?        Ssl  12:41   0:00 /opt/iojs-v2.3.4-linux-x64/bin/iojs /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/node_modules/auto-daemon/run.js /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/server.js --sockfile /opt/autoexec/sock --autoclose true --parentpid 15064 -- --sockfile /opt/autoexec/sock --pidfile /opt/autoexec/psy.pid --statefile /opt/autoexec/state --autoclose
root     15104  0.3  2.5 667856 26568 ?        Ssl  12:41   0:00 /opt/iojs-v2.3.4-linux-x64/bin/iojs /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/node_modules/auto-daemon/run.js /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/server.js --sockfile /opt/autoexec/sock --autoclose true --parentpid 15095 -- --sockfile /opt/autoexec/sock --pidfile /opt/autoexec/pid --statefile /opt/autoexec/state --autoclose
root     15136  0.3  2.5 667856 26364 ?        Ssl  12:41   0:00 /opt/iojs-v2.3.4-linux-x64/bin/iojs /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/node_modules/auto-daemon/run.js /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/server.js --sockfile /opt/autoexec/sock --autoclose true --parentpid 15122 -- --sockfile /opt/autoexec/sock --pidfile /opt/autoexec/pid --statefile /opt/autoexec/state --autoclose
root     15185  0.3  2.5 667856 26512 ?        Ssl  12:41   0:00 /opt/iojs-v2.3.4-linux-x64/bin/iojs /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/node_modules/auto-daemon/run.js /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/server.js --sockfile /opt/autoexec/sock --autoclose true --parentpid 15176 -- --sockfile /opt/autoexec/sock --pidfile /opt/autoexec/pid --statefile /opt/autoexec/state --autoclose
root     15379  0.3  2.5 667856 26504 ?        Ssl  12:42   0:00 /opt/iojs-v2.3.4-linux-x64/bin/iojs /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/node_modules/auto-daemon/run.js /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/server.js --sockfile /opt/autoexec/sock --autoclose true --parentpid 15370 -- --sockfile /opt/autoexec/sock --pidfile /opt/autoexec/pid --statefile /opt/autoexec/state --autoclose
root     16946  2.8  2.4 667592 25560 ?        Ssl  12:44   0:00 /opt/iojs-v2.3.4-linux-x64/bin/iojs /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/node_modules/auto-daemon/run.js /opt/iojs-v2.3.4-linux-x64/lib/node_modules/psy/server.js --sockfile /opt/autoexec/sock --autoclose true --parentpid 16937 -- --sockfile /opt/autoexec/sock --pidfile /opt/autoexec/pid --statefile /opt/autoexec/state --autoclose
root     17116  0.0  0.1  13552  1984 pts/1    S+   12:44   0:00 grep psy
```

```
root@node1:/opt/autoexec# which psy
/opt/iojs/bin/psy
root@node1:/opt/autoexec# which node
/opt/iojs/bin/node
```

Using iojs 2.3.4 and psy 1.6.2

**Comment by TehShrike** [comment]
After rolling back to 1.6.0 I do not observe the issue any more.

**Comment by acstll** [comment]
This is happening to me as well. I'm running node `4.2.3` via nvm.

I run `psy start -n foo -- node server/index.js` and it works.

`psy ls` works the first time, but then it shows nothing.

Every time I run `psy ls` I see a new `--autoclose` process as well.

I want to use `psy` in production but I just can't :-(

Is there anything I can do to provide you with useful information to spot a possible bug?

**Comment by danieljuhl** [comment]
I'm facing this issue too.. Running node@4.2.3 on Ubuntu 14.04

**Comment by maxogden** [comment]
i wrote a failing test for this here https://github.com/substack/psy/pull/13

**Comment by substack** [comment]
Sorry for the regression, this should be fixed now in 1.7.1.

**Comment by TehShrike** [comment]
<3
